spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -38,6 +38,16 @@
   end
 
   describe 'POST /api/saml/logout' do
+    context 'when there is an invalid SAML packet' do
+      let(:user) { create(:user, :signed_up) }
+
+      it 'responds with "400 Bad Request"' do
+        sign_in user
+
+        post :logout, params: { SAMLRequest: 'foo' }
+        expect(response.status).to eq(400)
```

Comment:
Should we also add a test for analytics to make sure it reports `saml_request_valid: false` and `sp_initiated: true`?

Reply:
I just noticed that the new one you added above is basically the same as this one. Can we combine the two, i.e. add `expect(response.status).to eq(400)` to the one above.